src/components/chart/state-history-chart-timeline.ts

```diff
@@ -221,9 +129,107 @@ export class StateHistoryChartTimeline extends LitElement {
       // If the line is more than 5 minutes old, re-gen it
       // so the X axis grows even if there is no new data
       this._generateData();
+      this._createOptions();
```

Comment:
This is the real change, refreshing the options when we've detected the data should change.